Modules/_localemodule.c

```diff
@@ -422,15 +422,18 @@ _locale_strxfrm_impl(PyObject *module, PyObject *str)
         goto exit;
     }
     if (n2 >= (size_t)n1) {
-        /* more space needed */
-        wchar_t * new_buf = PyMem_Realloc(buf, (n2+1)*sizeof(wchar_t));
+        /* more space needed, some implementations return needed size while
+           others just buffer length and it's up to the caller to figure
+           out needed buffer size */
+        size_t new_buf_len = n2 * 2;
```

Comment:
This code is working well for many years on many platforms. If only macOS 15 requires a change, would it be possible to make this code conditional on macOS? Something like:

```c
#ifdef __APPLE__
        // More space needed. macOS 15 just returns the buffer length and it's
        // up to the caller to figure out needed buffer size (gh-130567).
        size_t new_buf_len = n2 * 2;
#else
        size_t new_buf_len = n2 + 1;
#endif
```

Reply:
That might work, the thing is that this behavior is not defined in the specification, so it could behave this way elsewhere.

Reply:
If more platforms are impacted by this issue tomorrow, we can revisit the implementation later. For now, I would prefer to make the change specific to macOS (15).

Reply:
Okay, I've changed the code to use the return value in case it larger than the buffer and fall back to doubling if it is not.